dfastmi/batch/AreaDetector.py

```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+"""
+Copyright © 2024 Stichting Deltares.
+
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 2.1.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+import math
+from typing import Tuple
+
+import numpy
+
+
+class AreaDetector:
+
+    @property
+    def area(self) -> numpy.ndarray:
+        """
+        area
+        """
+        return self._area
+
+    @property
+    def volume(self) -> numpy.ndarray:
+        """
+        area volume
+        """
+        return self._volume
+
+    @property
+    def area_list(self) -> list:
+        """
+        List of sub areas
+        """
+        return self._area_list
+
+    @property
+    def total_area_weigth(self) -> numpy.ndarray:
+        """
+        total area weigth
+        """
+        return self._total_area_weigth
+
+    def __init__(self):
+        self._area: numpy.ndarray = numpy.zeros(0)
+        self._volume: numpy.ndarray = numpy.zeros(0)
+        self._area_list: list = []
+        self._total_area_weigth: numpy.ndarray = numpy.zeros(0)
+
+    def detect_areas(
+        self,
+        dzgemi: numpy.ndarray,
+        dzmin: float,
+        edgeface_indeces: numpy.ndarray,
+        areai: numpy.ndarray,
+        wbin: numpy.ndarray,
+        wthresh: numpy.ndarray,
+        siface: numpy.ndarray,
+        afrac: numpy.ndarray,
+        sbin: numpy.ndarray,
+        sthresh: numpy.ndarray,
+        slength: float,
+    ):
+        self._total_area_weigth = numpy.zeros(dzgemi.shape)
+
+        sbin_length = sthresh[1] - sthresh[0]
+        nwidth = wthresh[-1] - wthresh[0]
+        sub_areai, n_sub_areas = self._detect_connected_regions(
+            dzgemi > dzmin, edgeface_indeces
```

Comment:
Isnt the plural 'indexes' or 'indices'? :D

Reply:
fixed